### PR TITLE
Switched to ext_authz dynamic metadata in templates of Envoy 1.16+ and Istio 1.9

### LINKED ISSUE #163: [Template config] Change RBAC filter to use metadata; remove additional headers in ALS filter
Depending on the decision regarding https://github.com/apigee/apigee-remote-service-envoy/issues/149, we may want to remove RBAC filter entirely or comment it out.

### PR DESCRIPTION
**Pull request by rockspore**
`provision` command will set `append_metadata_headers` to true when creating config

`samples create` command will fail with error if the given config has improper value of `append_metadata_headers` with respect to the specified template.

fix #163

### CODE REVIEW

File: cmd/samples/samples_test.go [hunk 1]
```diff
@@ -274,6 +275,36 @@ func TestCreateIstioConfigsWithoutHttpbin(t *testing.T) {
...
+	defer os.Remove(tmpFile.Name())
+
+	// existing directory with no overwrite
+	rootArgs := &shared.RootArgs{}
+	flags := []string{"samples", "create", "--out", tmpDir, "-c", tmpFile.Name(), "-t", "istio-1.7"}
+	rootCmd := cmd.GetRootCmd(flags, print.Printf)
+	shared.AddCommandWithFlags(rootCmd, rootArgs, Cmd(rootArgs, print.Printf))
+
+	err = rootCmd.Execute()
+	testutil.ErrorContains(t, err, "checking config file's compatiblity with sample config: specified Istio/Envoy version requires append_metadata_headers to be true in the given config")
```
Comment (theganyo) [inline cmd/samples/samples_test.go:305]: typo: compatiblity

**Comment by rockspore** [comment]
We may eventually decide to remove the RBAC filter in the templates for Envoy 1.16 and Istio 1.9.

But the PR as of now at least can allow the integration builds to pass (by setting the `append_metadata_headers` to true).

File: templates/istio-1.9/envoyfilter-sidecar.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Installs an Istio EnvoyFilter in the default namespace.
+# It applies to all services in the namespace. If you wish, you
+# may tailor by specifying `workloadSelector` for specific targets.
+# Uses `apigee-remote-service-envoy.apigee:5000` as target.
+
+# This is for Istio 1.7+.
```
Comment (theganyo) [inline templates/istio-1.9/envoyfilter-sidecar.yaml:6]: should this be 1.9+?

**Comment by theganyo** [comment]
> We may eventually decide to remove the RBAC filter in the templates for Envoy 1.16 and Istio 1.9.

Should we add an issue?

File: templates/istio-1.9/envoyfilter-sidecar.yaml (hunk 2 above)
Comment (rockspore) [reply templates/istio-1.9/envoyfilter-sidecar.yaml]: Wonderful catch!

File: cmd/samples/samples_test.go (hunk 1 above)
Comment (rockspore) [reply cmd/samples/samples_test.go:305]: Thanks!

**Comment by rockspore** [comment]
> > We may eventually decide to remove the RBAC filter in the templates for Envoy 1.16 and Istio 1.9.
> 
> Should we add an issue?

Yes. Can do.

**Review by theganyo — DISMISSED** [review]

**Review by theganyo — APPROVED** [review]

**Pull request merged**